common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/OutLinedTextInput.kt

```diff
@@ -93,6 +94,17 @@ class OutLinedTextInputView @JvmOverloads constructor(
                     showKeyboard()
                 }
             }
+            getInt(R.styleable.OutLinedTextInputView_android_maxLines, -1).also {
+                if (it >= 0) {
+                    binding.internalEditText.maxLines = it
+                }
+            }
+
+            getInt(R.styleable.OutLinedTextInputView_android_inputType, InputType.TYPE_NULL).also {
```

Comment:
We are setting this already in `setupPasswordMode()` and `setupTextMode()`.  I think this currently overrides what the functions set. Do we only need this to allow singleLine only? 

I think we can either:
1. Retain android:inputType and remove setting inputType in the functions mentioned BUT we would need to manually set the inputType for password fields that we have now to have EditorInfo.TYPE_TEXT_VARIATION_PASSWORD.
2. Don't use android:inputType and just check if maxLines == 1 and pass a boolean to setupPasswordMode and setupTextMode to know if we need to set EditorInfo.TYPE_TEXT_FLAG_MULTI_LINE.


I think option 2 is better for our needs today.

Reply:
We could also just take `android:inputType` when the field is not a password / isPassword not set. Might be a simpler approach actually.

Reply:
@CDRussell How about setting a custom type with three modes? 

Something like:
```
<attr name="type" format="enum" >
    <enum name="multi_line" value="0"/>
    <enum name="single_line" value="1"/>
    <enum name="password" value="2"/>
</attr>
```
Each mode has it's own configuration, caters for all usages, and has multi_line as the default type.

I really don't like exposing inputType and maxLines as a public API

Reply:
@malmstein good suggestion! That should work as well and we can even  remove `app:IsPassword` and just rely on this `type`.

Just out of curiosity, what is the down side of exposing `inputType` and `maxLines`?

Reply:
I like that approach @malmstein , thanks. will give that a try.

>what is the down side of exposing inputType and maxLines?

Curious about this too. on the one hand, it's sort of leaking implementation details to expose the underlying attributes on the `EditText`. on the other hand, seems like a pain in having to redefine the wheel to allow all existing `EditText` properties to be configurable.

Reply:
Thanks @malmstein I agree! I think when creating custom components I am always torn between making components work for as much as possible configurations vs having a defined set of capabilities. I think your explanation above make sense as adding more possible customisations could result to breakage/inconsistencies.

Reply:
Switched approach as per suggestion above. Thanks!